### ISSUE TITLE
getCurrentSiteLocale // getPrimarySiteLocale

### ISSUE THREAD

**Issue by bradleyw92** [craftcms/cms#8019]
Below you will see two functions, I'm successful in returning my current site locale name with the first:

```
{% set currentSiteLocale = craft.i18n.getCurrentLocale() %}
{{ currentSiteLocale.name }}
```

However, I'm also looking to get the primary site locale but that doesn't seem to be possible, I can only seem to return the ID.

```
{% set primarySiteLocale = craft.i18n.getPrimarySiteLocale() %}
{{ primarySiteLocale }}
```

Is there any way around this at all?

**Comment by brandonkelly** [comment]
Both of those function are deprecated.

- Instead of `craft.i18n.getCurrentLocale()`, use `craft.app.locale`.
- Instead of `craft.i18n.getPrimarySiteLocale()`, use `craft.app.i18n.primarySiteLocale`.

That said, both of those deprecated methods and both of their replacements will give you a locale object, not the ID.

The reason you are seeing a difference is because in your first example, you are treating it as an object and outputting its `name` property (which is also deprecated in favor of `getDisplayName()` by the way), whereas in the second you are outputting the locale object directly, which will invoke its `__toString()` method, which outputs its ID.

In other words, if you add a `.name` to the second example, you will see similar behavior as the first example:

```twig
{% set primarySiteLocale = craft.i18n.getPrimarySiteLocale() %}
{{ primarySiteLocale.name }}
```

While you’re at it, you should switch to the non-deprecated syntax:

```twig
{% set currentSiteLocale = craft.app.locale %}
{{ currentSiteLocale.getDisplayName(craft.app.language) }}
```

```twig
{% set primarySiteLocale = craft.app.i18n.primarySiteLocale %}
{{ primarySiteLocale.getDisplayName(craft.app.language) }}
```
